File: Logica de Programação/Subalgoritmo/Listas.py
# Cria uma lista heterogênea com diferentes tipos de dados
lista = ["a", 1, 7, 2, True, 4.5]
print(lista)  # Exibe a lista inicial

# Adiciona o valor 45 no final da lista
lista.append(45)
print(lista)

# Insere o valor 56 na posição 0 (início da lista)
lista.insert(0, 56)
print(lista)

# Remove o último elemento da lista
lista.pop()
print(lista)

# Remove o elemento que está na penúltima posição (índice -3)
lista.pop(-3)
print(lista)

# Limpa todos os elementos da lista, deixando-a vazia
lista.clear()

# Loop que roda 5 vezes para adicionar números informados pelo usuário na lista
for cont in range(0, 5, 1):
    x = float(input("Digite um numero: "))  # Lê um número (float) do usuário
    lista.append(x)  # Adiciona esse número na lista
    print(lista)  # Mostra a lista atualizada a cada iteração

"""
# Este bloco está comentado, mas ele serviria para exibir os elementos da lista 
# usando acesso por índice.
for i in range(0, 5, 1):
    print(lista[i])
"""

# Percorre a lista exibindo cada elemento diretamente, sem usar índices
for elem in lista:
    print(elem)

# Calcula a soma dos elementos da lista
soma = 0
for elem in lista:
    soma += elem
print("\nsoma = ", soma)

File: Logica de Programação/Subalgoritmo/Listas.java
import java.util.Scanner; // Importa a classe Scanner para ler dados do teclado

public class Listas {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in); // Cria um objeto Scanner para entrada de dados

        int vetor[] = new int[5]; // Declara um vetor de inteiros com 5 posições
        int i; // Declara uma variável de controle para os loops

        System.out.println("Preenchendo o vetor...");
        // Loop para preencher o vetor com valores informados pelo usuário
        for (i = 0; i < 5; i++) {
            System.out.print("Posição [" + i + "] = ");
            vetor[i] = entrada.nextInt(); // Lê um número inteiro e armazena na posição i do vetor

            entrada.close(); // <- ERRO: Fechar o Scanner dentro do loop faz com que ele feche na primeira execução. 
                             // Isso gera erro nas próximas leituras. O correto seria fechar após todos os usos, no final do programa.
        }

        System.out.println("\nExibindo o vetor...");
        // Loop para exibir os valores armazenados no vetor
        for (i = 0; i < 5; i++) {
            System.out.println("Vetor [" + i + "] = " + vetor[i]);
        }

        int soma;
        soma = 0; // Inicializa a variável soma com 0

        // Loop para somar todos os elementos do vetor
        for (i = 0; i < 5; i++) {
            soma = soma + vetor[i]; // Soma acumulando os valores do vetor
        }

        // Exibe a soma total dos valores do vetor
        System.out.println("\nSoma = " + soma);
    }
}
